### PR TITLE
local build script to build the docker image

### PR DESCRIPTION
**Pull request by ydubreuil**
Simulate the Jenkins Pipeline docker step locally in a shell script so BlueOcean is built like in Jenkins

@reviewbybees

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 node {
   deleteDir()
   checkout scm
-  
+
```
Comment (michaelneale) [inline Jenkinsfile]: whitespace changes for the Jenkinsfile are a little annoying

**Comment by michaelneale** [comment]
Can you update this to master (next week) and try again?

**Comment by ydubreuil** [comment]
@michaelneale rebased on master and removed the irrelevant `Jenkinsfile` modifications

**Comment by michaelneale** [comment]
I like it. Is build.sh the right name? If so, 🐝  from me - otherwise a better name would be also good.

**Comment by michaelneale** [comment]
@ydubreuil hrm... I think this should tie into also building a complete docker image - which is https://github.com/jenkinsci/blueocean-plugin/pull/244 - being fixed. Perhaps this script should also have an option to do that? WDYT?

**Comment by ydubreuil** [comment]
@michaelneale yes, building an image in the end makes total sense, actually, I did it in the first place before removing it while pushing. I didn't want to build an image every time. I'll add an option to build the image on demand :)

**Comment by tfennelly** [comment]
Cool stuff @ydubreuil !!! So this is building a docker image? Maybe the name of the script should reflect that? Also, a few comments would help too I think e.g. if there are command line params ... examples of how to run it etc etc

**Comment by michaelneale** [comment]
yeah this is most useful - increasingly I like the idea of being able to reproduce things with simple scripts as much as possible. For some value of "simple script" ;)

seems some minor conflicts now

**Comment by ydubreuil** [comment]
Addressed most comments I hope, there's a online help now which explains what this script is for. And rebased against master

I didn't try it on MacOS, and I'm afraid it can fails because it requires to share workspace folder with the build container (same requirement as the Pipeline Docker plugin btw).

**Comment by ydubreuil** [comment]
Um, regarding the script name, it can probably be renamed `build-in-docker.sh`

**Comment by michaelneale** [comment]
@ydubreuil it would likely have better success with the new "docker for OSX beta" which does proper bind mounting (unlike the virtual box based ones) - docker pipeline certainly works well on osx now with that. 

Yes, go ahead and rename it, why not!

**Comment by ydubreuil** [comment]
@michaelneale script renamed, could you merge?

**Comment by michaelneale** [comment]
🐝

**Comment by michaelneale** [comment]
thanks @ydubreuil. Done.

**Pull request merged**